node/resolvers/config.ts

```diff
@@ -19,121 +19,123 @@ export const getAppId = (): string => {
   return appName
 }
 
-const checkConfig = async (ctx: Context) => {
-  const {
-    vtex: { logger },
-    clients: { mail, masterdata, vbase },
-  } = ctx
+class Config {
```

Comment:
In JS, functions are generally preferred to classes, except in a few cases where having instances really helps. In this case, we could just keep the function like before.

My suspicion is that you changed this to a class because of troubles mocking the `checkConfig` function, right? When using jest mocks this is feasible though, you can use the [`jest.mock` function](https://jestjs.io/docs/jest-object#jestmockmodulename-factory-options) in your test file, which will automatically make all imports for this file receive a mocked version of it instead.

So, in your test file you would do something like this:

```ts
import checkConfig from '../config'

jest.mock('../config')

// The following line is just so you can access jest.fn functions on `mockedCheckConfig`, otherwise
// Typescript will think it's the real checkConfig function instead.
const mockedCheckConfig = checkConfig as ReturnType<typeof jest.fn>

// In your test code you can mock the return value
mockedCheckConfig.mockReturnValue('mockedResult')
// Or the whole implementation
mockedCheckConfig.mockImplementation((param1, param2) => { // any logic at all here})
```

Reply:
I tried to rollback the implementation of `checkConfig` and use the mock, but i failure in all attempts...